pkg/cache/tas_flavor_snapshot.go

```diff
@@ -388,7 +409,12 @@ func (s *TASFlavorSnapshot) findLevelWithFitDomains(levelIdx int, required bool,
 func (s *TASFlavorSnapshot) updateCountsToMinimum(domains []*domain, count int32) []*domain {
 	result := make([]*domain, 0)
 	remainingCount := count
-	for _, domain := range domains {
+	for i, domain := range domains {
+		if !features.Enabled(features.LeastAllocatedTAS) {
+			mostAllocatedIdx := findMostAllocatedDomainIdx(domains[i:], remainingCount)
```

Comment:
I might be missing something here. I think this loop above `for i, domain := range domains` combined with linear `findMostAllocatedDomainIdx` makes me think again about complexity. Imagine you have 1000 nodes, all empty, and you have a big job spanning all of them, it would be quadratic, no?

I was assuming in my other comment that the `findMostAllocatedDomainIdx` is run only to find a replacement for the last domain in the chain (when remainingCount drops to 0, or would drop to 0 when accounting for the domain).

Reply:
Yes, that's true. We can prevent that by calling the `findMostAllocatedDomainIdx` only for the last domain in the chain. I'm on it

Reply:
I see, you are right, because remainingCoubt is still large for all the domains except last. 

I think this is awfully tricky calling the function and checking the condition inside the inner loop. Maybe it is enough to move the condition outside the inner loop or add comment.

Reply:
I agree, updated the logic

Reply:
Actually, it might be brain teaser, but I think when the function used binary-search the complexity of choosing the domains after sorting was O(nlogn), while now it is O(n). (in total it is still O(nlogn) because we sort the domains)..